------------------------------------------------------------
revno: 14958
revision-id: squid3@treenet.co.nz-20161226021522-gh5su0gpeuhgn33n
parent: squid3@treenet.co.nz-20161216041412-mgxm925beqc22340
fixes bug: http://bugs.squid-cache.org/show_bug.cgi?id=3940
committer: Amos Jeffries <squid3@treenet.co.nz>
branch nick: 4
timestamp: Mon 2016-12-26 15:15:22 +1300
message:
  Bug 3940 pt2: Make 'cache deny' do what is documented
  
  Instead of overriding whatever cacheability was previously set to
  (including changing non-cacheables to be cacheable) actually
  prevent both cache read and write.
  
  This 'regression' appears after initial bug 3940 fix.
------------------------------------------------------------
# Bazaar merge directive format 2 (Bazaar 0.90)
# revision_id: squid3@treenet.co.nz-20161226021522-gh5su0gpeuhgn33n
# target_branch: http://bzr.squid-cache.org/bzr/squid3/trunk/
# testament_sha1: c89ecec253ec9f2a60398e0ef34444dc010a9616
# timestamp: 2016-12-26 02:23:05 +0000
# source_branch: http://bzr.squid-cache.org/bzr/squid3/4
# base_revision_id: squid3@treenet.co.nz-20161216041412-\
#   mgxm925beqc22340
# 
# Begin patch
=== modified file 'src/client_side_request.cc'
--- a/src/client_side_request.cc	2016-11-04 16:47:34 +0000
+++ b/src/client_side_request.cc	2016-12-26 02:15:22 +0000
@@ -1399,7 +1399,10 @@
 ClientRequestContext::checkNoCacheDone(const allow_t &answer)
 {
     acl_checklist = NULL;
-    http->request->flags.cachable = (answer == ACCESS_ALLOWED);
+    if (answer == ACCESS_DENIED) {
+        http->request->flags.noCache = true; // dont read reply from cache
+        http->request->flags.cachable = false; // dont store reply into cache
+    }
     http->doCallouts();
 }
 

